libsentrykube/tests/test_service.py

```diff
@@ -510,3 +511,47 @@ def mock_truediv(self, other):
     ):
         with pytest.raises(Exception, match="Invalid YAML"):
             get_service_flags("test-service")
+
+
+def test_merge_config_typical():
```

Comment:
Can we add some tests demonstrating the merging functionality?

eg

```
dict_a = {foo: bar, baz: {qux: qux2}}
dict_b = {a: b, foo: foo2}
```

And assert various outputs from the different strategies? Would help us understand how they're expected to behave.